compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -262,36 +166,14 @@ impl DepNodeExt for DepNode {
     /// refers to something from the previous compilation session that
     /// has been removed.
     fn extract_def_id<'tcx>(&self, tcx: TyCtxt<'tcx>) -> Option<DefId> {
-        if self.kind.fingerprint_style(tcx) == FingerprintStyle::DefPathHash {
+        if tcx.fingerprint_style(self.kind) == FingerprintStyle::DefPathHash {
             Some(tcx.def_path_hash_to_def_id(DefPathHash(self.hash.into()), &mut || {
                 panic!("Failed to extract DefId: {:?} {}", self.kind, self.hash)
             }))
         } else {
             None
         }
     }
-
-    /// Used in testing
-    fn from_label_string(
```

Comment:
As those are specific to the list of `DepKind`s and only used in tests, I'd rather keep them in here.